src/mut/section.cpp

```diff
@@ -22,33 +32,38 @@ Section::Section(Morphology* morphology,
     , _id(id_)
     , _sectionType(type_) {}
 
-Section::Section(Morphology* morphology, unsigned int id_, const morphio::Section& section_)
+Section::Section(std::weak_ptr<Morphology> morphology,
```

Comment:
Don t we want to .lock() the morphology in the constructor as a sanity check ? to avoid producing a section from a count ref = 0 morpho ?
Not sure if this can happen tho.

Reply:
(` .lock()`  plus if (shared) of course.